internal/provider/resource_instance.go

```diff
@@ -256,11 +257,16 @@ func (r *instanceResource) Schema(ctx context.Context, _ resource.SchemaRequest,
 			"external_ips": schema.SetNestedAttribute{
 				Optional:    true,
 				Description: "External IP addresses provided to this instance.",
+				Validators: []validator.Set{
+					instanceExternalIPValidator{},
+				},
 				NestedObject: schema.NestedAttributeObject{
 					Attributes: map[string]schema.Attribute{
 						"id": schema.StringAttribute{
 							Description: "If type is ephemeral, ID of the IP pool to retrieve addresses from, or all available pools if not specified. If type is floating, ID of the floating IP",
 							Optional:    true,
+							Computed:    true,
+							Default:     stringdefault.StaticString(""),
```

Comment:
The `Computed` and `Default` accounts for the case where an instance created with an external IP using the default IP pool (i.e., `id = null`) would drift when read (e.g., `id = ""`) and require updating in place.

Reply:
Thanks! Do you mind adding that as a comment for future reference?

Reply:
Done!